inspirehep/dojson/utils/__init__.py

```diff
@@ -196,21 +198,50 @@ def encode_for_marcxml(value):
 def strip_empty_values(obj):
     """Recursively strips empty values."""
     if isinstance(obj, dict):
-        for key, value in obj.items():
-            value = strip_empty_values(value)
-            if value or value is False or value == 0:
-                obj[key] = value
+        for key, val in obj.items():
+            new_val = strip_empty_values(val)
+            # There are already lots of leaks in the callers of this function,
+            # there's no need to make it idempotent and double memory.
```

Comment:
@jacquerie that's why I return `obj or None` and `type(obj)(new_obj) or None`.  So it's the same behavior of returning `None` for everything that is either a truthy value or `0` or `False`